src/interpreter/stepper.rs

```diff
@@ -0,0 +1,214 @@
+use super::{
+    CachedMir,
+    TerminatorTarget,
+    ConstantId
+};
+use interpreter::function::FnEvalContext;
+use interpreter::global::GlobalEvalContext;
+use interpreter::frame::{FrameInterpreter, Frame};
+use error::EvalResult;
+use rustc::mir::repr as mir;
+use rustc::ty::subst::{self, Subst};
+use rustc::hir::def_id::DefId;
+use rustc::mir::visit::{Visitor, LvalueContext};
+use syntax::codemap::Span;
+use std::rc::Rc;
+
+pub enum Event {
+    Constant,
+    Assignment,
+    Terminator,
+    Done,
+}
+
+pub struct Stepper<'fncx, 'a: 'fncx, 'b: 'a + 'mir, 'mir: 'fncx, 'tcx: 'b>{
+    fncx: &'fncx mut FnEvalContext<'a, 'b, 'mir, 'tcx>,
+    mir: CachedMir<'mir, 'tcx>,
+    process: fn (&mut Stepper<'fncx, 'a, 'b, 'mir, 'tcx>) -> EvalResult<()>,
```

Comment:
Can we replace the function pointers with an `enum`? Instead of calling `(self.process)(self)`, match over the enum and call whatever method is needed.

I'm not sure why we'd use a function pointer in the first place.